cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java

```diff
@@ -1603,6 +1608,9 @@ private void showRemoteTimeseries(
       } catch (InterruptedException e) {
         logger.error("Interrupted when getting timeseries schemas in node {}.", node, e);
         Thread.currentThread().interrupt();
+      } finally {
+        // record the queried node to release resources later
+        ((RemoteQueryContext) context).registerRemoteNode(node, group.getHeader());
```

Comment:
@OneSizeFitsQuorum Here I reference the `ClusterReaderFactory.getRemoteGroupByExecutor()` to handle the `registerRemoteNode` to make sure the remote resource can be actually released after query.